tests/regression/diffsysctl/diffsysctl_test.py

```diff
@@ -0,0 +1,220 @@
+        self.old_module = old_module
+        self.new_module = new_module
+        self.result = result
+
+
+specs_path = "tests/regression/diffsysctl/test_specs"
+tasks_path = "tests/regression/diffsysctl/kernel_modules"
+
+
+def collect_task_specs():
+    """Collecting and parsing YAML files with test specifications."""
+    result = list()
+    if not os.path.isdir(tasks_path):
+        os.mkdir(tasks_path)
+    cwd = os.getcwd()
+    os.chdir(specs_path)
+    for spec_file_path in glob.glob("*.yaml"):
+        with open(spec_file_path, "r") as spec_file:
+            spec_yaml = yaml.load(spec_file)
+            if "disabled" in spec_yaml and spec_yaml["disabled"] is True:
+                continue
+            for sysctl in spec_yaml["sysctls"]:
+                proc_yaml = sysctl["proc_handler"]
+                data_yaml = sysctl["data_variable"]
+
+                spec = dict(spec_yaml)
+                spec["sysctl"] = sysctl["sysctl"]
+                spec["data"] = data_yaml["name"]
+                spec["data_functions"] = data_yaml["functions"]
+                spec["proc_handler"] = proc_yaml
+
+                spec_id = os.path.splitext(spec_file_path)[0] + "_" + \
+                    sysctl["sysctl"]
+                result.append((spec_id, spec))
+
+    os.chdir(cwd)
+    return result
+
+
+specs = collect_task_specs()
+
+
+class TaskSpec:
+    """
+    Task specification representing one testing scenario (one sysctl parameter)
+    """
+    def __init__(self, spec):
+        # Values from the YAML file
+        self.old_kernel = spec["old_kernel"]
+        self.new_kernel = spec["new_kernel"]
+        self.sysctl = spec["sysctl"]
+        self.data = spec["data"]
+        self.data_functions = list()
+        for function, result in spec["data_functions"].iteritems():
+            self.data_functions.append(FunctionSpec(function, result))
+        self.proc_handler = list()
+        for function, result in spec["proc_handler"].iteritems():
+            self.proc_handler.append(FunctionSpec(function, result))
+        if "control_flow_only" in spec:
+            self.control_flow_only = spec["control_flow_only"]
+        else:
+            self.control_flow_only = False
+        self.debug = spec["debug"] if "debug" in spec else False
+
+
+def prepare_task(spec):
+    """
+    Prepare testing task (scenario). Build old and new modules and copy
+    created files.
+    """
+    function_list = spec.proc_handler + spec.data_functions
+
+    # Find the modules
+    first_builder = LlvmKernelBuilder(spec.old_kernel, None,
+                                      debug=spec.debug)
+    second_builder = LlvmKernelBuilder(spec.new_kernel, None,
+                                       debug=spec.debug)
+
+    sysctl_module = first_builder.build_sysctl_module(spec.sysctl)
+    spec.data_diffsysctl = sysctl_module.get_data(spec.sysctl)
+    spec.proc_handler_diffsysctl = \
+        sysctl_module.get_proc_fun(spec.sysctl)
```

Comment:
I think we should call get_data and get_proc_fun in test_llvm_sysctl_module, not here.

Reply:
get_proc_fun is OK to move to test_llvm_sysctl_module, but the variable spec.data_diffsysctl has to be created in the prepare_task function, because functions_diff needs a KernelParam object with the indices inside it - this is not in the spec file, therefore it has to be extracted using LlvmSysctlModule from the sysctl itself.